File: modules/secure-recorder/ios/StopReason.swift
import Foundation

/**
 * Reason why recording was stopped for iOS/iPadOS
 * 
 * Represents the reason a recording session was stopped. Provides conversion to/from
 * JavaScript string format for Expo module communication.
 * 
 * iOS/iPadOS SPECIFICITY:
 * - Swift enum (case-based)
 * - camelCase naming (.durationLimit, .fileSizeLimit, .userStopped, .error)
 */
enum StopReason {
  case durationLimit
  case fileSizeLimit
  case userStopped
  case error
  
  /**
   * Convert to string for JavaScript/TypeScript compatibility
   */
  internal func toJsString() -> String {
    switch self {
    case .durationLimit:
      return "duration_limit"
    case .fileSizeLimit:
      return "file_size_limit"
    case .userStopped:
      return "user_stopped"
    case .error:
      return "error"
    }
  }
  
  /**
   * Parse from JavaScript string (for backward compatibility)
   */
  internal static func fromJsString(_ value: String) -> StopReason {
    switch value {
    case "duration_limit":
      return .durationLimit
    case "file_size_limit":
      return .fileSizeLimit
    case "user_stopped":
      return .userStopped
    case "error":
      return .error
    default:
      return .error // Default to error for unknown values
    }
  }
}

File: modules/secure-recorder/android/src/main/java/expo/modules/securerecorder/StopReason.kt
package expo.modules.securerecorder

/**
 * Reason why recording was stopped for Android
 * 
 * Represents the reason a recording session was stopped. Provides conversion to/from
 * JavaScript string format for Expo module communication.
 * 
 * ANDROID SPECIFICITY:
 * - Kotlin enum class
 * - UPPER_SNAKE_CASE naming (DURATION_LIMIT, FILE_SIZE_LIMIT, USER_STOPPED, ERROR)
 */
enum class StopReason {
  DURATION_LIMIT,
  FILE_SIZE_LIMIT,
  USER_STOPPED,
  ERROR;
  
  /**
   * Convert to string for JavaScript/TypeScript compatibility
   */
  internal fun toJsString(): String {
    return when (this) {
      DURATION_LIMIT -> "duration_limit"
      FILE_SIZE_LIMIT -> "file_size_limit"
      USER_STOPPED -> "user_stopped"
      ERROR -> "error"
    }
  }
  
  companion object {
    /**
     * Parse from JavaScript string (for backward compatibility)
     */
    internal fun fromJsString(value: String): StopReason {
      return when (value) {
        "duration_limit" -> DURATION_LIMIT
        "file_size_limit" -> FILE_SIZE_LIMIT
        "user_stopped" -> USER_STOPPED
        "error" -> ERROR
        else -> ERROR // Default to ERROR for unknown values
      }
    }
  }
}
